test/e2e/framework/kubernetes/create-agnhost-statefulset.go

```diff
@@ -0,0 +1,157 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+var ErrLabelMissingFromPod = fmt.Errorf("label missing from pod")
+
+const (
+	AgnhostHTTPPort = 80
+	AgnhostReplicas = 1
+)
+
+type CreateAgnhostStatefulSet struct {
```

Comment:
you can drop the "Create" prefix ? Just AgnhostStatefulSet is good ?

Reply:
Also will be easy to have a New() func ?

Reply:
> you can drop the "Create" prefix ? Just AgnhostStatefulSet is good ?

Since is this an actional step and not an actual appsv1.StatefulSet struct, I'd prefer to keep verb prefix, that way when using intellisense and I type `kubernetes.` I'd see all of the create steps adjacent to each other in that package matching a similar verb prefix

```
kubernetes.CreateAgnhostStatefulSet
kubernetes.CreateNeworkPolicy
Kubernetes.CreateX
```

and then it's easy to track in the test when to have a corresponding step doing another crud action

```
kubernetes.DeleteAgnhostStatefulSet
kubernetes.DeleteNetworkPolicy
...
```

Reply:
> Also will be easy to have a New() func

imo adding the struct directly to the slice makes it very readable to see what parameters are passed, if using a NewObjectName(...), then it opens the door for users of the step to misplace the order of the parameters, and then might end up creating a resource with an unintended name

See: https://github.com/Azure/azure-container-networking/pull/2527/files#diff-649efb346a9d03496b0347a2dfd0214ff4f242b3ae8ef54ca3ca02b00931a43fR32-R53